tonic/models/vic/grid_params.py

```diff
@@ -1356,6 +1618,246 @@ def write_netcdf(myfile, target_attrs, target_grid,
+    v.units = unit.soil_param['run_cell']
+    v.description = desc.soil_param['run_cell']
+    for var in ['lats', 'lons']:
+        v = f.createVariable(var, NC_DOUBLE, dims2, fill_value=FILLVALUE_F)
+        v[:, :] = soil_grid[var]
+        if var == 'lats':
+            v.long_name = 'latitude of grid cell center'
+        elif var == 'lons':
+            v.long_name = 'longitude of grid cell center'
+        v.units = unit.soil_param[var]
+        v.description = desc.soil_param[var]
+
+    # State variables
+    for var, data in state_grid.items():
+        print('writing var: {0} {1}'.format(var, data.shape))
+
+        if state_grid[var].ndim == 2:
+            if var in ['Lake_snow_age', 'Lake_snow_melt_state',
+                       'Lake_active_layers']:
+                v = f.createVariable(var, NC_INT, dims2,
+                                     fill_value=FILLVALUE_I)
+            else:
+                v = f.createVariable(var, NC_DOUBLE, dims2,
+                                     fill_value=FILLVALUE_F)
+            v[:, :] = data
+
+        elif state_grid[var].ndim == 3:
+            if var == 'Lake_soil_moisture':
+                mycoords = ('nlayer', ) + dims2
+            elif var == 'Lake_soil_node_temp':
+                mycoords = ('soil_node', ) + dims2
+            elif var in ['Lake_layer_surf_area', 'Lake_layer_temp']:
+                mycoords = ('lake_node', ) + dims2
+            v = f.createVariable(var, NC_DOUBLE, mycoords,
+                                 fill_value=FILLVALUE_F)
+            v[:, :, :] = data
+
+        elif state_grid[var].ndim == 4:
+            if var == 'Lake_soil_ice':
+                mycoords = ('nlayer', 'frost_area', ) + dims2
+            else:
+                mycoords = ('veg_class', 'snow_band', ) + dims2
+            if var in ['Snow_age', 'Snow_melt_state']:
+                v = f.createVariable(var, NC_INT, mycoords,
+                                     fill_value=FILLVALUE_I)
+            else:
+                v = f.createVariable(var, NC_DOUBLE, mycoords,
+                                     fill_value=FILLVALUE_F)
+            v[:, :, :, :] = data
+
+        elif state_grid[var].ndim == 5:
+            if var == 'Soil_moisture':
+                mycoords = ('veg_class', 'snow_band', 'nlayer', ) + dims2
+            elif var == 'Soil_node_temp':
+                mycoords = ('veg_class', 'snow_band', 'soil_node', ) + dims2
+            v = f.createVariable(var, NC_DOUBLE, mycoords,
+                                 fill_value=FILLVALUE_F)
+            v[:, :, :, :, :] = data
+
+        elif state_grid[var].ndim == 6:
+            mycoords = ('veg_class', 'snow_band', 'nlayer',
+                        'frost_area', ) + dims2
+            v = f.createVariable(var, NC_DOUBLE, mycoords,
+                                 fill_value=FILLVALUE_F)
+            v[:, :, :, :, :, :] = data
+        elif var in ['dz_node','node_depth']:
+            continue
```

Comment:
I forget why I handled these variables differently.  Maybe I can put them back into that loop.  Otherwise, I'll add a comment explaining why they're different.